app/src/main/java/com/vci/vectorcamapp/core/data/cache/DeviceCacheImplementation.kt

```diff
@@ -23,25 +26,45 @@ class DeviceCacheImplementation @Inject constructor(
     }
 
     override suspend fun getDevice(): Device? {
-        val deviceCacheDto = dataStore.data.firstOrNull()
-        return if (deviceCacheDto == null || deviceCacheDto == DeviceCacheDto()) {
+        val dto = dataStore.data.first()
+        return if (dto.programId == -1) {
             null
         } else {
             Device(
-                id = deviceCacheDto.id,
-                model = deviceCacheDto.model,
-                registeredAt = deviceCacheDto.registeredAt,
-                submittedAt = deviceCacheDto.submittedAt
+                id = dto.id,
+                model = dto.model,
+                registeredAt = dto.registeredAt,
+                submittedAt = dto.submittedAt
             )
         }
     }
 
     override suspend fun getProgramId(): Int? {
```

Comment:
Can you explain why you decided to migrate to using the program ID instead of checking to see if the device cache dto is null?